setup.py

```diff
@@ -70,81 +70,37 @@ def load_version_module(pkg_path):
         # Minimum jaxlib version; used in testing.
         'minimum-jaxlib': [f'jaxlib=={_minimum_jaxlib_version}'],
 
-        # CPU-only jaxlib can be installed via:
-        # $ pip install jax[cpu]
-        'cpu': [f'jaxlib=={_current_jaxlib_version}'],
+        # A CPU-only jax doesn't require any extras, but we keep this extra
+        # around for compatibility.
+        'cpu': [],
 
         # Used only for CI builds that install JAX from github HEAD.
         'ci': [f'jaxlib=={_latest_jaxlib_version_on_pypi}'],
 
         # Cloud TPU VM jaxlib can be installed via:
         # $ pip install jax[tpu] -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
         'tpu': [
-          f'jaxlib=={_current_jaxlib_version}',
+          f'jaxlib>={_current_jaxlib_version},<={_jax_version}',
           f'libtpu-nightly=={_libtpu_version}',
           'requests',  # necessary for jax.distributed.initialize
         ],
 
-        # CUDA installations require adding the JAX CUDA releases URL, e.g.,
-        # Cuda installation defaulting to a CUDA and Cudnn version defined above.
-        # $ pip install jax[cuda] -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html
-        'cuda': [f"jaxlib=={_current_jaxlib_version}+cuda12.cudnn{_default_cuda12_cudnn_version}"],
-
-
-      'cuda12_pip': [
```

Comment:
Anyone with `pip install jax[cuda12_pip]` in their CI will get an error once this releases... should we keep this for backward compatibility?

Reply:
Hmm. But then what's the path to remove it? There's no way we can gently deprecate this, right?

Reply:
Readded in https://github.com/google/jax/pull/21851

Reply:
Not sure... I don't think there's any way to gracefully remove this.